packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -120,6 +174,55 @@ export default class SaytDriverPlugin implements Plugin {
     return [searchTerms];
   }
 
+  /**
+   * Extracts query and products from the given response.
+   *
+   * @param response An object containing the original query and product records.
+   * @returns An object containing the query and an array of valid simplified products.
+   */
+  searchCallback(response: Results): ProductsResponseSection {
+    const { query, records } = response;
+    const mappedRecords = records.map(record => {
+      const filter = this.parseRecord(record);
+      if (filter === undefined) return;
+      const { data, firstVariant, nonvisualVariants } = filter;
+      return {
```

Comment:
We'll have to turn this into a product transformer later.

Reply:
I'd assumed this would have to be more nuanced, but can you elaborate on what you mean?

Reply:
By this, I meant that this transformation is specific to our sample dataset and that we'll have to make it more generic later.